libcob/common.h

```diff
@@ -1432,6 +1432,7 @@ typedef struct __cob_file {
 	char				*nxt_filename;	/* Next position in org_filename */
 	unsigned int		flag_is_concat:1;	/* SEQUENTIAL concatenated file names */
 
+	const unsigned char* code_set_write;	/* CODE-SET conversion for RE-/WRITE) */
```

Comment:
If you want to do this splitting - which seems reasonable - then you need to check the module version and "move" the field for GC 3.2 generated modules, as those will have used the field above. Also you'd have to move it at the end of the `cob_file`, as hinted in the doc-comment for `__cob_file`.

Note: If you _always_ place the key collation into _each_ `cob_key`, then you don't need to have the file-collation in `cob_file` at all (if set then those will never be empty).